lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -809,15 +804,9 @@ private function convertSingleBooleanValue($value, callable $callback)
             return $callback(true);
         }
 
-        /**
-         * Better safe than sorry: http://php.net/in_array#106319
-         */
-        if (in_array(strtolower(trim($value)), $this->booleanLiterals['false'], true)) {
-            return $callback(false);
-        }
-
-        if (in_array(strtolower(trim($value)), $this->booleanLiterals['true'], true)) {
-            return $callback(true);
+        $boolean = self::BOOLEAN_LITERALS[strtolower(trim($value))] ?? null;
+        if ($boolean !== null) {
```

Comment:
This `?? null` followed by `!== null` hides the intent of the code. Previously, it was clear: `if (in_array(/* ... */))` and it must remain the same clear: `if (isset(/* ... */))`.